src/services/motion/types.ts

```diff
@@ -72,7 +72,7 @@ export const encodeMotionConfiguration = (
   const result = new pb.MotionConfiguration();
 
   result.setObstacleDetectorsList(
-    obj.obstacleDetectorsList.map((od: ObstacleDetector) => {
+    obj.obstacleDetectorsList?.map((od: ObstacleDetector) => {
```

Comment:
This is a bit hard to read . Instead I'd recommend:
```ts
const { obstacleDetectorsList = [] } = obj;
result.setObstacleDetectorsList(obstacleDetectorsList.map(...))
```
We can then remove the `new Array` coalescing below on line 84.